### PR TITLE
Fix/city is required in interrogation of contacts

### PR DESCRIPTION
**Pull request by chenFriedman**

### CODE REVIEW

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningInfo/ContactDetails.tsx [hunk 1]
```diff
@@ -16,10 +16,14 @@ const ContactDetails = (props: Props) => {
...
     const { index , interactedContact } = props;
 
+    const isStatusCompleted = interactedContact.contactStatus === 5;
+
     const formErrors = errors.form ? (errors.form[index] ? errors.form[index] : {}) : {};
     const formHasErrors = Object.entries(formErrors)
         .some(([key, value]) => value !== undefined);
 
+    const showErrorIcon = formHasErrors && !isStatusCompleted;
+
```
Comment (y-golde) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningInfo/ContactDetails.tsx:26]: Change the method to validate in the schema itself, look inside the schema for id or phone validation and how it's done there, this approach is not scalable, contact me if you need help

Comment (y-golde) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningInfo/ContactDetails.tsx:26]: Meaning ignore the validation if the status is 5

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactSection/Schemas/contactQuestioningClinical.ts [hunk 2]
```diff
@@ -8,4 +8,7 @@ export const contactQuestioningClinical = {
     [InteractedContactFields.DOES_NEED_HELP_IN_ISOLATION]: yup
         .boolean()
         .nullable(),
+    [InteractedContactFields.ISOLATION_ADDRESS]: yup.object().shape({
+        'city': yup.string().nullable().required('שגיאה: שדה חובה')
+    })
```
Comment (y-golde) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactSection/Schemas/contactQuestioningClinical.ts:13]: Add a when statement here like phone number or id

**Review by y-golde — CHANGES_REQUESTED** [review]

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningInfo/ContactDetails.tsx [hunk 3]
```diff
@@ -16,10 +16,14 @@ const ContactDetails = (props: Props) => {
     const { errors } = useFormContext();
     const { index , interactedContact } = props;
 
+    const isStatusCompleted = interactedContact.contactStatus === 5;
```
Comment (barnahum8) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningInfo/ContactDetails.tsx:19]: change the 5 value to const

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/InteractedContactAccordion.tsx [hunk 4]
```diff
@@ -36,6 +36,8 @@ const InteractedContactAccordion = (props: Props) => {
         shouldDisable,
     } = props;
 
+    const isStatusCompleted = interactedContact.contactStatus === 5
```
Comment (barnahum8) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/InteractedContactAccordion.tsx:39]: here as well

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningInfo/ContactDetails.tsx (hunk 3 above)
Comment (chenFriedman) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningInfo/ContactDetails.tsx]: I solved this case in the schema so I removed it...

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/InteractedContactAccordion.tsx (hunk 4 above)
Comment (chenFriedman) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/InteractedContactAccordion.tsx]: I solved this case in the schema so I removed it...

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningInfo/ContactDetails.tsx (hunk 1 above)
Comment (chenFriedman) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningInfo/ContactDetails.tsx]: fixed [here](client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactSection/Schemas/contactQuestioningClinical.ts)

Comment (chenFriedman) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningInfo/ContactDetails.tsx]: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactSection/Schemas/contactQuestioningClinical.ts

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactSection/Schemas/contactQuestioningClinical.ts [hunk 5]
```diff
@@ -8,4 +8,12 @@ export const contactQuestioningClinical = {
     [InteractedContactFields.DOES_NEED_HELP_IN_ISOLATION]: yup
         .boolean()
         .nullable(),
+    [InteractedContactFields.ISOLATION_ADDRESS]: yup.object()
+    .when(InteractedContactFields.CONTACT_STATUS, {
+        is: 5,
```
Comment (barnahum8) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactSection/Schemas/contactQuestioningClinical.ts:13]: use enum 'ContactStatusCodes'

**Review by barnahum8 — APPROVED** [review]

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningInfo/ContactDetails.tsx (hunk 1 above)
Comment (chenFriedman) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningInfo/ContactDetails.tsx]: fixed here : client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactSection/Schemas/contactQuestioningClinical.ts

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactSection/Schemas/contactQuestioningClinical.ts (hunk 2 above)
Comment (chenFriedman) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactSection/Schemas/contactQuestioningClinical.ts]: fixed here : client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactSection/Schemas/contactQuestioningClinical.ts

**Review by y-golde — APPROVED** [review]

**Review by barnahum8 — APPROVED** [review]

**Pull request merged**